cognite_toolkit/_cdf_tk/utils/auth.py

```diff
@@ -59,741 +32,446 @@
+    PROVIDER: Provider = field(default="entra_id", metadata=EnvOptions("Provider", "entra_id"))
+    CDF_URL: str | None = field(
         default=None,
-        metadata=dict(env_name="CDF_URL", display_name="CDF URL", example="https://CDF_CLUSTER.cognitedata.com"),
-    )
-    login_flow: LoginFlow = field(
-        default="client_credentials",
-        metadata=dict(
-            env_name="LOGIN_FLOW",
-            display_name="Login flow",
-            example="client_credentials",
-        ),
+        metadata=EnvOptions("CDF URL", "https://{CDF_CLUSTER}.cognitedata.com", optional=frozenset(ALL_CASES)),
     )
-    provider: Provider = field(
-        default="entra_id",
-        metadata=dict(
-            env_name="PROVIDER",
-            display_name="Provider",
-            example="entra_id",
-        ),
-    )
-    token: str | None = field(
-        default=None, metadata=dict(env_name="CDF_TOKEN", display_name="OAuth2 token", example="")
+    CDF_TOKEN: str | None = field(
+        default=None, metadata=EnvOptions("OAuth2 token", example="", required=frozenset([("token", None)]))
     )
-    client_id: str | None = field(
+    IDP_CLIENT_ID: str | None = field(
         default=None,
-        metadata=dict(
-            env_name="IDP_CLIENT_ID", display_name="client id", example="XXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX"
+        metadata=EnvOptions(
+            display_name="client id",
+            example="",
+            required=frozenset([("client_credentials", None), ("interactive", None), ("device_code", "other")]),
         ),
     )
-    client_secret: str | None = field(
+    IDP_CLIENT_SECRET: str | None = field(
         default=None,
-        metadata=dict(env_name="IDP_CLIENT_SECRET", display_name="client secret", example=""),
+        metadata=EnvOptions(
+            display_name="client secret", example="", is_secret=True, required=frozenset([("client_credentials", None)])
+        ),
     )
-    token_url: str | None = field(
+    IDP_TOKEN_URL: str | None = field(
         default=None,
-        metadata=dict(
-            env_name="IDP_TOKEN_URL",
+        metadata=EnvOptions(
             display_name="token URL",
-            example="https://login.microsoftonline.com/IDP_TENANT_ID/oauth2/v2.0/token",
+            example="https://login.microsoftonline.com/{IDP_TENANT_ID}/oauth2/v2.0/token",
+            required=frozenset([("client_credentials", "other")]),
+            optional=frozenset([("client_credentials", "entra_id")]),
         ),
     )
-    tenant_id: str | None = field(
+    IDP_TENANT_ID: str | None = field(
         default=None,
-        metadata=dict(
-            env_name="IDP_TENANT_ID",
+        metadata=EnvOptions(
             display_name="Tenant id for MS Entra",
-            example="12345678-1234-1234-1234-123456789012",
+            example="mytenant.onmicrosoft.com",
```

Comment:
We are looking for the tenant GUID here

```suggestion
            example="00000000-0000-0000-0000-000000000000",
```